src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [ ];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const value of keysToRemove) {
+          delete stateCopy[value];
+        }
+        break;
+
+      case 'clear' :
+        for (const key of Object.keys(stateCopy)) {
+          delete stateCopy[key];
+        }
+        break;
+
+      default:
+        result.push({});
+        break;
```

Comment:
don't need to push empty object to result 
```suggestion
      default:
        break;
```